### PR TITLE
MIP 10.2.4

### PR DESCRIPTION
**Pull request by jemten**
### This PR adds | fixes:

- Hotfix to add tempdirecory for Star-Fusion in an attempt to avoid timeout, also includes a RNA resource bump

### How to test:

- Automatic and continuous test suite

### Expected outcome:
- [x] Installation, unit and integration test suite pass

### Review:
- [x] Code review
- [x] Tests pass

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes. If applicable record manual test results in PR header
- [ ] **MINOR** - when you add functionality in a backwards compatible manner. If applicable record manual test results in PR header
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: lib/MIP/Program/Star_fusion.pm [hunk 1]
```diff
@@ -35,13 +35,16 @@ sub star_fusion {
...
+## Arguments: $aln_temp_directory     => Temporary directory
+##          : $cpu                    => Number of threads for running STAR
 ##          : $examine_coding_effect  => Append coding effect to fusion
 ##          : $fastq_r1_path          => The path of the R1 fastq
 ##          : $fastq_r2_path          => The path of the R2 fastq
 ##          : $filehandle             => Filehandle to write to
 ##          : $fusion_inspector       => Run FusionInspector in either inspect or validate mode
+##          : $fusion_temp_directory  => Temporary directory
 ##          : $genome_lib_dir_path    => Path to the directory containing the genome library
+##          :
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Star_fusion.pm:47]:
```suggestion
```

File: lib/MIP/Recipes/Analysis/Glnexus.pm [hunk 2]
```diff
@@ -190,13 +190,16 @@ sub analysis_glnexus {
 
     say {$filehandle} q{## } . $recipe_name;
 
+    ## Limit memory allocation for glnexus
+    Readonly my $FIFTY => 50;
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Analysis/Glnexus.pm:194]:
```suggestion
    Readonly my $REDUCE_GLNEXUS_MEMORY => 50;
```
or something similar.

File: lib/MIP/Recipes/Analysis/Star_aln.pm [hunk 3]
```diff
@@ -744,4 +744,363 @@ sub analysis_star_aln_mixed {
     return 1;
 }
 
+sub analysis_star_fusion_aln {
+
+## Function : Alignment of fastq files using star aln
+## Returns  :
+## Arguments: $active_parameter_href   => Active parameters for this analysis hash {REF}
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Analysis/Star_aln.pm:751]:
```suggestion
## Arguments: $active_parameter_href => Active parameters for this analysis hash {REF}
```
And for the rest of the args

**Review by henrikstranneheim — APPROVED** [review]
👍 ! Had to put my Perl glasses on - it's been a while.

File: lib/MIP/Recipes/Analysis/Star_aln.pm (hunk 3 above)
Comment (jemten) [reply lib/MIP/Recipes/Analysis/Star_aln.pm]: fixed

**Comment by jemten** [comment]
Deepvariant part tested on justhusky (DNA)
```
#jobid	jobname	account	partition	alloccpus	TotalCPU	elapsed	start	end	state	exitcode
2696680	deepvariant_earlycasualcaiman	developme+	core	36	4-01:23:03	04:55:58	2022-03-09T23:19:44	2022-03-10T04:15:42	COMPLETED	0:0
2696681	deepvariant_hugelymodelbat	developme+	core	36	4-03:07:39	05:05:25	2022-03-09T23:22:10	2022-03-10T04:27:35	COMPLETED	0:0
2696682	deepvariant_slowlycivilbuck	developme+	core	36	4-01:04:34	05:03:51	2022-03-09T23:25:48	2022-03-10T04:29:39	COMPLETED	0:0
2696683	glnexus_merge_justhusky	developme+	core	32	01:03:58	00:21:47	2022-03-10T04:29:39	2022-03-10T04:51:26	COMPLETED	0:0
2696684	gatk_baserecalibration_earlycasualcaiman	developme+	core	26	07:48:05	00:31:13	2022-03-09T23:55:19	2022-03-10T00:26:32	COMPLETED	0:0
2696685	gatk_baserecalibration_hugelymodelbat	developme+	core	26	07:38:45	00:30:30	2022-03-10T00:23:29	2022-03-10T00:53:59	COMPLETED	0:0
2696686	gatk_baserecalibration_slowlycivilbuck	developme+	core	26	07:05:24	00:29:26	2022-03-10T01:10:22	2022-03-10T01:39:48	COMPLETED	0:0
2696687	gatk_haplotypecaller_earlycasualcaiman	developme+	core	16	1-15:18:54	03:41:35	2022-03-10T00:30:34	2022-03-10T04:12:09	COMPLETED	0:0
2696688	gatk_haplotypecaller_hugelymodelbat	developme+	core	16	1-14:14:04	03:35:29	2022-03-10T01:01:51	2022-03-10T04:37:20	COMPLETED	0:0
2696689	gatk_haplotypecaller_slowlycivilbuck	developme+	core	16	1-13:52:41	03:33:42	2022-03-10T01:40:04	2022-03-10T05:13:46	COMPLETED	0:0
2696690	gatk_genotypegvcfs_justhusky	developme+	core	3	01:11:37	01:33:58	2022-03-10T05:13:47	2022-03-10T06:47:45	COMPLETED	0:0
2696691	gatk_genotypegvcfs_justhusky	developme+	core	3	01:15:11	01:39:06	2022-03-10T05:13:47	2022-03-10T06:52:53	COMPLETED	0:0
2696692	gatk_genotypegvcfs_justhusky	developme+	core	3	59:57.905	01:16:48	2022-03-10T05:13:47	2022-03-10T06:30:35	COMPLETED	0:0
2696693	gatk_genotypegvcfs_justhusky	developme+	core	3	58:21.679	01:14:09	2022-03-10T05:13:47	2022-03-10T06:27:56	COMPLETED	0:0
2696694	gatk_genotypegvcfs_justhusky	developme+	core	3	54:20.275	01:09:10	2022-03-10T05:13:47	2022-03-10T06:22:57	COMPLETED	0:0
2696695	gatk_genotypegvcfs_justhusky	developme+	core	3	56:25.446	01:12:51	2022-03-10T05:13:47	2022-03-10T06:26:38	COMPLETED	0:0
2696696	gatk_genotypegvcfs_justhusky	developme+	core	3	50:45.226	01:03:46	2022-03-10T05:13:47	2022-03-10T06:17:33	COMPLETED	0:0
2696697	gatk_genotypegvcfs_justhusky	developme+	core	3	48:51.128	01:02:47	2022-03-10T05:13:47	2022-03-10T06:16:34	COMPLETED	0:0
2696698	gatk_genotypegvcfs_justhusky	developme+	core	3	37:33.007	00:48:11	2022-03-10T05:13:47	2022-03-10T06:01:58	COMPLETED	0:0
2696699	gatk_genotypegvcfs_justhusky	developme+	core	3	38:33.864	00:54:15	2022-03-10T05:13:47	2022-03-10T06:08:02	COMPLETED	0:0
2696700	gatk_genotypegvcfs_justhusky	developme+	core	3	38:36.511	00:52:46	2022-03-10T05:13:47	2022-03-10T06:06:33	COMPLETED	0:0
2696701	gatk_genotypegvcfs_justhusky	developme+	core	3	39:14.972	00:49:59	2022-03-10T05:13:47	2022-03-10T06:03:46	COMPLETED	0:0
2696702	gatk_genotypegvcfs_justhusky	developme+	core	3	28:54.718	00:38:58	2022-03-10T05:13:47	2022-03-10T05:52:45	COMPLETED	0:0
2696703	gatk_genotypegvcfs_justhusky	developme+	core	3	26:21.002	00:34:12	2022-03-10T05:13:47	2022-03-10T05:47:59	COMPLETED	0:0
2696704	gatk_genotypegvcfs_justhusky	developme+	core	3	25:08.327	00:31:23	2022-03-10T05:13:47	2022-03-10T05:45:10	COMPLETED	0:0
2696705	gatk_genotypegvcfs_justhusky	developme+	core	3	27:08.888	00:33:39	2022-03-10T05:13:47	2022-03-10T05:47:26	COMPLETED	0:0
2696706	gatk_genotypegvcfs_justhusky	developme+	core	3	26:59.670	00:33:55	2022-03-10T05:13:47	2022-03-10T05:47:42	COMPLETED	0:0
2696707	gatk_genotypegvcfs_justhusky	developme+	core	3	26:33.898	00:34:44	2022-03-10T05:13:47	2022-03-10T05:48:31	COMPLETED	0:0
2696708	gatk_genotypegvcfs_justhusky	developme+	core	3	00:21:14	00:26:44	2022-03-10T05:13:47	2022-03-10T05:40:31	COMPLETED	0:0
2696709	gatk_genotypegvcfs_justhusky	developme+	core	3	19:03.512	00:23:57	2022-03-10T05:13:47	2022-03-10T05:37:44	COMPLETED	0:0
2696710	gatk_genotypegvcfs_justhusky	developme+	core	3	11:21.542	00:12:57	2022-03-10T05:13:47	2022-03-10T05:26:44	COMPLETED	0:0
2696711	gatk_genotypegvcfs_justhusky	developme+	core	3	11:39.605	00:13:45	2022-03-10T05:13:47	2022-03-10T05:27:32	COMPLETED	0:0
2696712	gatk_genotypegvcfs_justhusky	developme+	core	3	46:52.330	01:02:21	2022-03-10T05:13:47	2022-03-10T06:16:08	COMPLETED	0:0
2696713	gatk_genotypegvcfs_justhusky	developme+	core	3	04:38.807	00:04:27	2022-03-10T05:13:47	2022-03-10T05:18:14	COMPLETED	0:0
2696714	gatk_genotypegvcfs_justhusky	developme+	core	3	00:45.956	00:00:36	2022-03-10T05:13:47	2022-03-10T05:14:23	COMPLETED	0:0
2696715	gatk_gathervcfs_justhusky	developme+	core	2	01:16.538	00:01:21	2022-03-10T06:53:19	2022-03-10T06:54:40	COMPLETED	0:0
2696716	gatk_variantrecalibration_justhusky	developme+	core	6	38:23.211	00:36:04	2022-03-10T06:54:40	2022-03-10T07:30:44	COMPLETED	0:0
2696717	gatk_combinevariantcallsets_justhusky	developme+	core	5	02:46.111	00:02:05	2022-03-10T07:30:47	2022-03-10T07:32:52	COMPLETED	0:0
2696718	qccollect_ar_justhusky	developme+	core	1	00:05.641	00:00:11	2022-03-10T07:32:53	2022-03-10T07:33:04	COMPLETED	0:0
2696719	version_collect_ar_justhusky	developme+	core	2	01:13.395	00:02:20	2022-03-10T07:33:04	2022-03-10T07:35:24	COMPLETED	0:0
2696720	multiqc_ar_justhusky	developme+	core	1	01:11.232	00:01:48	2022-03-10T07:35:24	2022-03-10T07:37:12	COMPLETED	0:0
```

Starfusion part tested on easybaboon (RNA)
```
#jobid	jobname	account	partition	alloccpus	TotalCPU	elapsed	start	end	state	exitcode
2696723	star_fusion_ACC9368A1	developme+	core	6	04:01:07	07:46:25	2022-03-09T18:27:38	2022-03-10T02:14:03	COMPLETED	0:0
2696724	arriba_ar_ACC9368A1	developme+	core	24	11:35:49	00:48:55	2022-03-10T02:14:15	2022-03-10T03:03:10	COMPLETED	0:0
2696725	megafusion_ar_ACC9368A1	developme+	core	1	00:04.248	00:00:16	2022-03-10T03:03:10	2022-03-10T03:03:26	COMPLETED	0:0
2696726	fusion_report_ACC9368A1	developme+	core	6	11:03.980	00:11:33	2022-03-10T03:03:26	2022-03-10T03:14:59	COMPLETED	0:0
2696727	svdb_merge_fusion_easybaboon	developme+	core	1	00:01.775	00:00:07	2022-03-10T03:25:58	2022-03-10T03:26:05	COMPLETED	0:0
2696728	merge_fusion_reports_easybaboon	developme+	core	2	03:35.742	00:03:43	2022-03-10T03:27:14	2022-03-10T03:30:57	COMPLETED	0:0
2696729	qccollect_ar_easybaboon	developme+	core	1	00:22.325	00:00:36	2022-03-10T03:31:30	2022-03-10T03:32:06	COMPLETED	0:0
2696730	version_collect_ar_easybaboon	developme+	core	2	01:16.109	00:02:31	2022-03-10T03:32:37	2022-03-10T03:35:08	COMPLETED	0:0
2696731	multiqc_ar_easybaboon	developme+	core	2	00:32.859	00:00:54	2022-03-10T03:35:16	2022-03-10T03:36:10	COMPLETED	0:0
```

**Review by ramprasadn — APPROVED** [review]
LGTM! 👍🏻

**Pull request merged**